crates/sign/src/lib.rs

```diff
@@ -51,9 +51,20 @@ pub fn verify_hash(hash: Hash, signature: &Signature) -> Result<(), secp256k1::E
 /// Verify the given message against the given signature.
 fn verify_message(msg: &Message, signature: &Signature) -> Result<(), secp256k1::Error> {
```

Comment:
I wonder if we should remove this now as it's wrong anyway. This does a recover then a verify for no reason

Reply:
Do we not still want a way to verify when the pubkey is not known?

Reply:
oh nvm. my brain wasn't working. Yeah I can remove this and just replace with the new fn.

Reply:
It can be swapped out to use `recover_hash` or `recover_from_message`